src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -563,6 +597,27 @@ async Task<PurgeHistoryResult> IDurableOrchestrationClient.PurgeInstanceHistoryA
             return await this.DurabilityProvider.PurgeInstanceHistoryByInstanceId(instanceId);
         }
 
+        async Task<PurgeHistoryResult> IDurableOrchestrationClient.PurgeInstanceHistoryAsync(IEnumerable<string> instanceId)
+        {
+            var tasks = new List<Task>();
+            int instanceDeleted = 0;
+            foreach (string id in instanceId)
+            {
+                tasks.Add(Task.Run(async () =>
+                {
+                    var purgeHistoryResult = await this.DurabilityProvider.PurgeInstanceHistoryByInstanceId(id);
+                    if (purgeHistoryResult.InstancesDeleted == 1)
+                    {
+                        instanceDeleted += 1;
+                    }
+                }));
+            }
+
+            await Task.WhenAll(tasks);
```

Comment:
Hmm, the combination of `Task.Run` and `Task.WhenAll` in this block of code doesn't quite seem right to me.
My understanding is that, when you call `Task.Run`, you're asking for that `Task` to execute immediately.

However, that is also achieved in the `Task.WhenAll`: that also schedules all input tasks.

So that's all to say that I think this code could be simpler :)

Can you try the following:

When you create the list of `tasks`, create it as such:

```C#
tasks.add(this.DurabilityProvider.PurgeInstanceHistoryByInstanceId(id);
```

Then you'll have a list of Tasks that each returns a `PurgeHistoryResult`. From that list, you should be able to calculate the number of instances deleted :) . Let me know if you need with this.